mlir/lib/Catalyst/Transforms/DetectQNodes.cpp

```diff
@@ -748,6 +748,7 @@ void replaceTerminatorWithUnconditionalJumpToSuccessBlock(SmallVector<Block *> a
         PatternRewriter::InsertionGuard insertGuard(rewriter);
         auto terminator = abort->getTerminator();
         rewriter.setInsertionPoint(terminator);
+        assert(isa<LLVM::UnreachableOp>(terminator));
```

Comment:
This change affects the logic of the program and must be placed in a different commit that is not documentation